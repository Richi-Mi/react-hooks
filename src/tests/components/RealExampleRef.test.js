import { shallow } from "enzyme"
import RealExampleRef from "../../components/04-useRef/RealExampleRef"

describe(' * Pruebas en el Componente <RealExampleRef /> ', () => {
    const wrapper = shallow( <RealExampleRef /> );
    test(' - Debe de Mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
        
    });
    test(' - Debe de Mostrar el componente <MultipleCustomHooks />', () => {

        const btn_show = wrapper.find('button');
        btn_show.simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
    
    });
});
