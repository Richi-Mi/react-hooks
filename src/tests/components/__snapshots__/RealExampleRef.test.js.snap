// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas en el Componente <RealExampleRef />   - Debe de Mostrarse correctamente 1`] = `
<div>
  <h1>
     Real Example Ref 
  </h1>
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
